app/main.py

```diff
@@ -111,36 +144,41 @@ def get_post(id: int):
 # Brisanje posta koji ima uneti id
 # Dodatno brisanje pretrazenog id-a iz baze podataka
 @app.delete("/posts/{id}", status_code=status.HTTP_204_NO_CONTENT)
-def delete_post(
-    id: int,
-):
-    cursor.execute("""DELETE FROM posts where id = %s  RETURNING *""", str(id))
-    deleted_post = cursor.fetchone()
-    conn.commit()
-    if deleted_post == None:
+def delete_post(id: int, db: Session = Depends(get_db)):
+    # cursor.execute("""DELETE FROM posts where id = %s  RETURNING *""", str(id))
+    # deleted_post = cursor.fetchone()
+    # conn.commit()
+    deleted_post = db.query(models.Post).filter(models.Post.id == id)
+    if deleted_post.first() == None:
```

Comment:
Potential null pointer exception can happen if deleted_post is null (None), so check if it is possible to happen.